src/entity-manager/EntityManager.ts

```diff
@@ -1354,8 +1354,8 @@ export class EntityManager {
      * sets current EntityManager instance to it. Used to work with custom repositories
      * in transactions.
      */
-    withRepository<Entity extends ObjectLiteral, R extends Repository<Entity>>(
-        repository: R,
+    withRepository<Entity extends ObjectLiteral, R extends Repository<any>>(
```

Comment:
This gives the right type inference with the new ObjectLiteral constraint

Reply:
`any`? Shouldn't it be `ObjectLiteral` too?

Reply:
it is constrained in the argument itself (line 1358), otherwise the type inference doesn't work